code/edgemedia/src/main/java/com/adobe/marketing/mobile/edge/media/internal/MediaXDMEventHelper.kt

```diff
@@ -0,0 +1,266 @@
+                        MediaConstants.VideoMetadataKeys.DAY_PART -> sessionDetails.dayPart = value
+                        MediaConstants.VideoMetadataKeys.EPISODE -> sessionDetails.episode = value
+                        MediaConstants.VideoMetadataKeys.FEED -> sessionDetails.feed = value
+                        MediaConstants.VideoMetadataKeys.FIRST_AIR_DATE -> sessionDetails.firstAirDate = value
+                        MediaConstants.VideoMetadataKeys.FIRST_DIGITAL_DATE -> sessionDetails.firstDigitalDate = value
+                        MediaConstants.VideoMetadataKeys.GENRE -> sessionDetails.genre = value
+                        MediaConstants.VideoMetadataKeys.MVPD -> sessionDetails.mvpd = value
+                        MediaConstants.VideoMetadataKeys.NETWORK -> sessionDetails.network = value
+                        MediaConstants.VideoMetadataKeys.ORIGINATOR -> sessionDetails.originator = value
+                        MediaConstants.VideoMetadataKeys.RATING -> sessionDetails.rating = value
+                        MediaConstants.VideoMetadataKeys.SEASON -> sessionDetails.season = value
+                        MediaConstants.VideoMetadataKeys.SHOW -> sessionDetails.show = value
+                        MediaConstants.VideoMetadataKeys.SHOW_TYPE -> sessionDetails.showType = value
+                        MediaConstants.VideoMetadataKeys.STREAM_FORMAT -> sessionDetails.streamFormat = value
+
+                        // Audio standard metadata cases
+                        MediaConstants.AudioMetadataKeys.ALBUM -> sessionDetails.album = value
+                        MediaConstants.AudioMetadataKeys.ARTIST -> sessionDetails.artist = value
+                        MediaConstants.AudioMetadataKeys.AUTHOR -> sessionDetails.author = value
+                        MediaConstants.AudioMetadataKeys.LABEL -> sessionDetails.label = value
+                        MediaConstants.AudioMetadataKeys.PUBLISHER -> sessionDetails.publisher = value
+                        MediaConstants.AudioMetadataKeys.STATION -> sessionDetails.station = value
+                    }
+                }
+            }
+
+            return sessionDetails
+        }
+
+        @JvmStatic
+        fun generateMediaCustomMetadata(metadata: Map<String, String>): List<XDMCustomMetadata> {
+            val customMetadataList = mutableListOf<XDMCustomMetadata>()
+
+            metadata.forEach { (key, value) ->
+                run {
+                    if (!standardMediaMetadataSet.contains(key)) {
+                        customMetadataList.add(XDMCustomMetadata(key, value))
+                    }
+                }
+            }
+
+            return customMetadataList.sortedBy { it.name }
+        }
+
+        @JvmStatic
+        fun generateAdvertisingPodDetails(adBreakInfo: AdBreakInfo?): XDMAdvertisingPodDetails? {
+            val adBreakInfo = adBreakInfo
+            if (adBreakInfo !is AdBreakInfo) {
```

Comment:
This technically works, but you are checking the type of the object in order to smart cast it to a non-optional. Instead, I'd suggest using the safe call operator `?.` to unwrap the object. You can do this in all places where are casting the object using `!is`

```
adBreakInfo?.let {
   return XDMAdvertisingPodDetails(it.name, it.position, it.startTime.toLong())
}

return null
```